### PR TITLE
Fips dsa keygen disable

### PR DESCRIPTION
**Pull request by slontis**
##### Checklist

- [x] documentation is added or updated
- [x] tests are added or updated

### CODE REVIEW

**Comment by slontis** [comment]
This is dependent on the FIPS DSA signing PR https://github.com/openssl/openssl/pull/24799.
Disregard the first commit if reviewing.
The other commits are related to DSA keygen

File: test/evp_test.c
Comment (nhorman) [inline test/evp_test.c:142]: style check says this line should align 8 chars in, rather than underneath the start of the arguments.  Not sure if you want to fix that or just waive the style

**Comment by slontis** [comment]
Short answer is No. There are multiple other PR's that are going to conflict with that PR.

**Comment by slontis** [comment]
(I also didnt want to add this to the other PR because it was too much in the one PR)

**Comment by slontis** [comment]
rebased with code style fixups

File: test/evp_test.c
Comment (slontis) [reply test/evp_test.c]: moved it to the || which is still wrong according to the check :)

File: doc/man3/EVP_KEYMGMT.pod
Comment (paulidale) [inline doc/man3/EVP_KEYMGMT.pod:101]: _respectively_ at the end perhaps?

File: doc/man3/EVP_KEYMGMT.pod
Comment (paulidale) [inline doc/man3/EVP_KEYMGMT.pod:146]: Spurious blank line?

**Review by paulidale — APPROVED** [review]

**Comment by slontis** [comment]
Minor doc NIT fixes.

**Review by paulidale — APPROVED** [review]

**Comment by slontis** [comment]
ping @openssl/committers

**Comment by paulidale** [comment]
Re-ping @openssl/committers for second review please.

**Comment by mattcaswell** [comment]
> Re-ping @openssl/committers for second review please.

This is already approved!!

**Comment by paulidale** [comment]
Will merge when the extended tests pass.

**Comment by slontis** [comment]
PR/master seems like a strange combo

**Comment by t8m** [comment]
> PR/master seems like a strange combo

It is valid if you need to fix an issue with PR targeting older branches. Yeah, OTOH in many cases it will fail if run on PR for a master branch. Perhaps we could add some additional condition to skip it if the target branch of the PR is master.

**Comment by t8m** [comment]
Merged to the master branch. Thank you for your contribution.

**Pull request closed without merging**
